include/jsonrpccxx/client.hpp

```diff
@@ -86,7 +87,37 @@ namespace jsonrpccxx {
       } else if (v == version::v1) {
         j["params"] = nullptr;
       }
-      connector.Send(j.dump());
+      GetConnector().Send(j);
     }
   };
+
+  class JsonRpcClientJson
+  : public JsonRpcClientBase {
+  public:
+    JsonRpcClientJson(IClientJsonConnector &connector, version v) : JsonRpcClientBase(v), connector(connector) {}
+
+  protected:
+    virtual IClientJsonConnector &GetConnector() override { return connector; }
+
+  private:
+    IClientJsonConnector &connector;
+  };
+
+  class JsonRpcClient
+  : public JsonRpcClientBase
+  , private IClientJsonConnector {
+  public:
+    JsonRpcClient(IClientConnector &connector, version v) : JsonRpcClientBase(v), connector(connector) {}
+
+  protected:
+    IClientConnector &connector;
+    virtual IClientJsonConnector &GetConnector() override { return *this; }
+
+  private:
+
+    virtual json IClientJsonConnector::Send(const json &request) {
```

Comment:
The automated tast chokes on this one, but I rather wait for futher feedback before trying to fix it.